### DISCUSSION TITLE
Pagination doenst works

### DISCUSSION THREAD

**Discussion by datlechin** [livewire/livewire#7074] [7. Bugs (archived)] [answered]
### Livewire version

v3.0.8

### Which PHP version are you using?

PHP 8.2

### Repository URL

private

### Steps To Reproduce

1. create new route with dynamic slug
```php
Route::get('categories/{category:slug}', CategoriesList::class);
```

In `CategoriesList` component:
```
class CategoriesList extend Component
{
   use WithPagination;
}
```
2. Then go to view add `$categories->links()`
3. Then click to paginate item, it will return 404

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by PhiloNL** [comment, marked as answer]
Hi @datlechin,
It seems to work, I've created an example based on your instructions:
https://wirebox.app/b/4omey

**Reply by datlechin** [reply]
Thank you. Now it works with a computed property.

Here is my previous code that didn't work with the `WithPagination` trait. Can you take a look to find where the issue is that's preventing it from working?


```php
    public function render(): View
    {
        $this->category = AccountCategory::query()
            ->where('slug', $this->category)
            ->active()
            ->firstOrFail();

        $accounts = $this->category
            ->accounts()
            ->where('status', AccountStatus::Selling)
            ->latest()
            ->with(['category', 'store'])
            ->paginate(12);

        return view('livewire.pages.accounts-list', ['accounts' => $accounts]);
    }
```

**Reply by PhiloNL** [reply]
I'm not sure, if I look at the code it should work I believe. I would recommend using computed properties when possible. It makes your component a bit cleaner (you can even remove the render method), and subsequent calls within the same request are cached.

**Reply by datlechin** [reply]
Thank you
